ansible-roles/k8s_fabric_peer/files/fabric-peer/templates/deployment.yaml

```diff
@@ -236,6 +239,65 @@ spec:
             - name: docker-tls
               mountPath: /docker-tls
         {{- end }}
+        {{- if .Values.fabricSnapshotArchiver.enabled }}
+        - name: fabric-snapshot-archiver
+          image: "{{ .Values.fabricSnapshotArchiver.image.repository }}:{{ .Values.fabricSnapshotArchiver.image.version }}"
+          imagePullPolicy: {{ .Values.fabricSnapshotArchiver.image.pullPolicy }}
+          command:
+            - /bin/sh
+            - -c
+            - while true; do sleep 10; done
+          env:
+            - name: AWS_REGION
+              value: {{ .Values.global.aws.region }}
+            - name: AWS_BUCKET
+              value: {{ .Values.global.aws.staticBucket }}
+            - name: AWS_BUCKET_PREFIX
+              value: "{{ .Values.global.lutherEnv }}/fabric-snapshots"
+            - name: FABRIC_CFG_PATH
+              value: /etc/hyperledger/fabric/
+            - name: FABRIC_LOGGING_SPEC
+              value: INFO
+            - name: CORE_PEER_ADDRESS
+              value: {{ include "fabric-peer.self-fqdn" . }}:{{ .Values.internal.container.gossipPort }}
+            - name: CORE_PEER_ID
+              value: fabric-snapshot-archiver
+            - name: CORE_PEER_LOCALMSPID
+              value: {{ .Values.dlt.mspID }}
+            - name: CORE_PEER_TLS_ENABLED
+              value: "true"
+            - name: CORE_PEER_TLS_CLIENTAUTHREQUIRED
+              value: "true"
+            - name: CORE_PEER_TLS_CERT_FILE
+              value: /etc/hyperledger/fabric/tls/server.crt
+            - name: CORE_PEER_TLS_KEY_FILE
+              value: /etc/hyperledger/fabric/tls/server.key
+            - name: CORE_PEER_TLS_CLIENTCERT_FILE
+              value: /etc/hyperledger/fabric/tls/server.crt
+            - name: CORE_PEER_TLS_CLIENTKEY_FILE
+              value: /etc/hyperledger/fabric/tls/server.key
+            - name: CORE_PEER_TLS_ROOTCERT_FILE
+              value: /etc/hyperledger/fabric/tls/ca.crt
+            - name: CORE_PEER_MSPCONFIGPATH
+              value: /etc/hyperledger/fabric/msp
+          volumeMounts:
+            - name: genesisblock
+              mountPath: /opt/gopath/src/github.com/hyperledger/fabric/peer/channel-artifacts
+            - name: orderertls
+              mountPath: /etc/hyperledger/fabric/orderertls
+            - name: config
+              subPath: core.yaml
+              mountPath: /etc/hyperledger/fabric/core.yaml
+            - name: adminmsp
+              mountPath: /etc/hyperledger/fabric/msp
+            - name: tls
+              mountPath: /etc/hyperledger/fabric/tls
+            - name: storage
+              mountPath: /var/hyperledger/production
```

Comment:
Does it need all of these? Perhaps not the artifacts or peer config... What is this admin MSP?

Reply:
yeah it does unfortunately. Remember the standard peer pod does not have an admin MSP and therefore does not have permission to issue `peer snapshot` command. This is another reason why we cannot run the snapshot command from the conventional peer pod.